packages/@ember/application/tests/readiness_test.js

```diff
@@ -110,26 +105,7 @@ moduleFor(
       });
 
       assert.equal(readyWasCalled, 1, 'ready was called now all readiness deferrals are advanced');
-    }
-
-    ["@test Application's ready event can be deferred by other components"](assert) {
-      jQuery.isReady = false;
-
-      run(() => {
-        application = Application.create({ router: false });
-        application.deferReadiness();
-        assert.equal(readyWasCalled, 0, "ready wasn't called yet");
-      });
-
-      domReady();
-
-      assert.equal(readyWasCalled, 0, "ready wasn't called yet");
-
-      run(() => {
-        application.advanceReadiness();
-      });
-
-      assert.equal(readyWasCalled, 1, 'ready was called now all readiness deferrals are advanced');
+      assert.equal(callbacks['DOMContentLoaded'].length, 0);
```

Comment:
@chancancode This test case is failing because of event listener added in `waitForDOMReady()`.
Should we remove it in `advanceReadiness()` method to not pollute the global state?

Reply:
Isn't the callback supposed to remove it once it is called?

Reply:
Indeed! Had to trigger 'DOMContentLoaded' in the test, since readyState should become `complete` eventually